### DISCUSSION TITLE
refresh voice status in author context

### DISCUSSION THREAD

**Discussion by Renaud-Dov** [Rapptz/discord.py#6876] [Q&A]
I use a predicate for a function that check if the author is connected in voice channel. I tried to use `context.author.voice`, but it looks like when i use the command for the first time and I am connected, it work, but if i leave the voice channel, for the bot, I'm still connected.
And it is the same if I was disconnected when I used it. When I try again but this time connected to a voice channel, and context.author.voice is still null
Is there any option ?

**Comment by Jonah987654321** [comment]
I had an similar error with voiceStates a few weeks ago, and for me it works, to let the bot edit a message before checking the voiceState. Idk why, but it helped for me
